SYCL/ESIMD/lsc/lsc_predicate.cpp

```diff
@@ -0,0 +1,200 @@
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include "../esimd_test_utils.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+using namespace sycl::ext::intel::experimental::esimd;
+
+int testAccessor(queue q) {
+  auto size = size_t{128};
+  auto constexpr SIMDSize = unsigned{4};
+
+  auto vec_0 = std::vector<int>(size);
+  auto vec_1 = std::vector<int>(size);
+  auto vec_2 = std::vector<int>(size);
+  auto vec_3 = std::vector<int>(size);
+
+  std::iota(vec_0.begin(), vec_0.end(), 0);
+  std::iota(vec_1.begin(), vec_1.end(), 0);
+  std::iota(vec_2.begin(), vec_2.end(), 0);
+  std::iota(vec_3.begin(), vec_3.end(), 0);
+  auto buf_0 = buffer{vec_0};
+  auto buf_1 = buffer{vec_1};
+  auto buf_2 = buffer{vec_2};
+  auto buf_3 = buffer{vec_3};
+
+  try {
+    q.submit([&](handler &h) {
+      auto access_0 = buf_0.template get_access<access::mode::read_write>(h);
+      auto access_1 = buf_1.template get_access<access::mode::read_write>(h);
+      auto access_2 = buf_2.template get_access<access::mode::read_write>(h);
+      auto access_3 = buf_3.template get_access<access::mode::read_write>(h);
+
+      h.parallel_for(
+          range<1>{size / SIMDSize}, [=](id<1> id) SYCL_ESIMD_KERNEL {
+            auto offset = id[0] * SIMDSize * sizeof(int);
+            auto offsets = simd<uint32_t, SIMDSize>(id * SIMDSize * sizeof(int),
+                                                    sizeof(int));
+            auto pred_enable = simd_mask<1>(1);
+            auto pred_disable = simd_mask<1>(0);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_0,
+                                                                   offset);
+            auto data_0 =
+                lsc_block_load<int, SIMDSize>(access_0, offset, pred_enable);
+            lsc_block_store<int, SIMDSize>(access_0, offset, data_0 * 2,
+                                           pred_enable);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_1,
+                                                                   offset);
+            auto data_1 =
+                lsc_block_load<int, SIMDSize>(access_1, offset, pred_disable);
+            lsc_block_store<int, SIMDSize>(access_1, offset, data_1 * 2,
+                                           pred_enable);
+
+            lsc_prefetch<int, SIMDSize, lsc_data_size::default_size,
+                         cache_hint::cached, cache_hint::uncached>(access_2,
+                                                                   offset);
+            auto data_2 =
+                lsc_block_load<int, SIMDSize>(access_2, offset, pred_enable);
+            lsc_block_store<int, SIMDSize>(access_2, offset, data_2 * 2,
```

Comment:
we have 4 loads and 4 stores - do they test some different aspects of the API? If not, please minimize.